### DISCUSSION TITLE
My first mesa project, and I want to contribute back

### DISCUSSION THREAD

**Discussion by ValerioB88** [mesa/mesa#1147] [Show and tell]
Hi, I have developed this Neuroevolution environment in Mesa
https://www.youtube.com/watch?v=pOF1H84xPik&feature=youtu.be&ab_channel=VBiscione

In order to do that, I have slightly changed the mesa codebase. I am a bit of a noob in collaboration workflow and was wondering what's the best way to contribute to the project or at least share my code.

My intuition is that the best way would be to fork the mesa repo, add my changes, and open a pull request (I believe the changes might be useful in general). 
Then, in a different repo, upload my code, which will somehow need to refer to my mesa fork (I am not sure how to do that... should I add my mesa-fork git repo in the requirements?)

Any suggestion is appreciated :) 

EDIT: this is now in a public repo https://github.com/ValerioB88/EvoAgent

**Comment by tpike3** [comment]
@ValerioB88 very cool!

If you follow this [guide](https://github.com/projectmesa/mesa/blob/main/CONTRIBUTING.rst) it should give you everything you need. let me know if you have any issues. Although we need to fix the hyperlinks as we no longer use google groups.

**Comment by aiwhoo** [comment]
@ValerioB88 Can you post a link to your github/fork as is? Your code demonstrates alot of capabilities that I've been wanting, but either are unclear from documentation or just not existing. 

Could you elaborate on what changes you've made?

**Reply by ValerioB88** [reply]
Hi @aiwhoo , I plan to release the code this week, maybe even today time permitting.
Mesa documentation is pretty good imho, so the features you see in my project are probably just not existing.

In the main repo I have changed a bit the html in order to add graphics on the sides of the page. I have change the max value of the frame rate slider. I have changed the runcontr.js in order to capture many shortcuts.
These are not many changes.

The rest is done separately as a new model. I fet around mesa for a lot of things: i have my own way of moving the agents in the env, perceicing the world, and saving data. 
I mostly use mesa for the websocket communication which is something I haven't ever done. 


For evolving the network I use neat-python.

**Reply by ValerioB88** [reply]
> @ValerioB88 Can you post a link to your github/fork as is? Your code demonstrates alot of capabilities that I've been wanting, but either are unclear from documentation or just not existing.
> 
> Could you elaborate on what changes you've made?

https://github.com/ValerioB88/EvoAgent/tree/master

**Reply by aiwhoo** [reply]
This is amazing!!!! Thank you so much!!!!!!

**Reply by tpike3** [reply]
@ValerioB88  very cool looking forward to the release
